src/mantarray_desktop_app/file_writer.py

```diff
@@ -196,8 +197,36 @@ def _drain_board_queues(
     return board_dict
 
 
+class FileDirMixIn(metaclass=abc.ABCMeta):
```

Comment:
Since DataAnalyzer isn't using this MixIn at the moment, we should probably get rid of it and just keep all this functionality in FileWriter itself for now